### PR TITLE
Create opening (#306)

### PR DESCRIPTION
**Pull request by traumschule**
According to #306

- [X] Add `Add Opening` button for lead to WG openings tab.
- [X] Open modal on click.
- [X] Import opening from json
- [X] Backup state to json
- [X] Reuse steps with form state  and state machine from`CreateWorkingGroupLeadOpening`
- [X] form watcher, Tx
- [X] CreateOpening.stories.tsx
- [ ] update steps and machine
- [ ] tests
- [ ] cleanup comments

[Discussion](https://discord.com/channels/811216481340751934/910968995379822633/1045411744236970095)

create-opening-button
opening

Selected file is automatically validated and content set on state, or error:
upload-validator-rror

Hint for local testing:
```
modified   packages/ui/src/app/pages/WorkingGroups/WorkingGroup/OpeningsTab.tsx
@@ -25,7 +25,8 @@ export const OpeningsTab = ({ workingGroup }: Props) => {                                                                                                                                                                    
   const { isLoading: isLoadingCurrent, openings } = useOpenings({ groupId: workingGroup.id, type: 'open' })                                                                                                                                   
   const { debt } = useGroupDebt(workingGroup.id)                                                                                                                                                                                              
   const { isLoading, workers } = useMyWorkers()                                                                                                                                                                                               
-  const isLead = useMemo(() => workers.find((w) => w.membership.id === workingGroup.leadId), [!isLoading])                                                                                                                                    
+  //const isLead = useMemo(() => workers.find((w) => w.membership.id === workingGroup.leadId), [!isLoading])                                                                                                                                  
+  const isLead = true
```

### CODE REVIEW

File: packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    'workingGroupAndDescription.title',
+    'workingGroupAndDescription.shortDescription',
+    'workingGroupAndDescription.description',
+    'durationAndProcess.details',
+    'durationAndProcess.duration',
+    'stakingPolicyAndReward.stakingAmount',
+    'stakingPolicyAndReward.leavingUnstakingPeriod',
+    'stakingPolicyAndReward.rewardPerBlock',
+  ])
+  **/
```
Comment (oleksanderkorn) [inline packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx:54]: Do we need this commented out block?

File: packages/ui/src/working-groups/modals/CreateOpening/components/Import.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }, [])
+
+  /**
+  useEffect(() => {
+    if (value?.content) {
+      handleChange(value.content)
+      onHide()
+    }
+  }, [value])
+**/
```
Comment (oleksanderkorn) [inline packages/ui/src/working-groups/modals/CreateOpening/components/Import.tsx:56]: Should this be cleaned up? :)

File: packages/ui/src/working-groups/modals/CreateOpening/machine.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+             * - DB:
+             * - O(1) doesn't depend on the state or parameters
+             * # </weight>
+
+            addOpening: AugmentedSubmittable<(
+              description: Bytes | string | Uint8Array,
+              openingType: PalletWorkingGroupOpeningType | 'Leader' | 'Regular' | number | Uint8Array,
+              stakePolicy: PalletWorkingGroupStakePolicy | { stakeAmount?: any; leavingUnstakingPeriod?: any; },
+              rewardPerBlock: Option<u128> | null | object | string | Uint8Array)
+              => SubmittableExtrinsic<ApiType>, [Bytes, PalletWorkingGroupOpeningType, PalletWorkingGroupStakePolicy, Option<u128>]>; **/
```
Comment (oleksanderkorn) [inline packages/ui/src/working-groups/modals/CreateOpening/machine.ts:37]: Same here, any reason of this being left?

**Review by oleksanderkorn — CHANGES_REQUESTED** [review]
LGTM, but i would say needs some cleanup.

**Comment by traumschule** [comment]
Thanks! It also needs help with the state machine, not sure when i get to it. :(

**Comment by oleksanderkorn** [comment]
As I can see in the last comment:

>@[dmtrjsg](https://github.com/dmtrjsg) dmtrjsg added this to Closed - On Prod in [Pioneer Community Product >Development](https://github.com/orgs/Joystream/projects/55#card-88069169) [2 days ago](https://github.com/Joystream/pioneer/pull/3891#event-8512582173)

@traumschule can we close this PR?

**Comment by traumschule** [comment]
>  can we close this PR?

In my mind only if there's a merged fix for #306. I was hoping for support by someone to create a PR on top of this one.

**Pull request closed without merging**